### PR TITLE
[PT FE]: add transformation for prim::min and TupleConstruct + ListConstruct

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 -  add *prim::min* support
 - *handle ListConstruct out in TupleConstruct out*
 - * update opset include to direct model includes in transformations

### Tickets:
 - *101124*

### CODE REVIEW

File: tests/layer_tests/pytorch_tests/test_tuple_construct.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2018-2022 Intel Corporation
```
Comment (eaidova) [inline tests/layer_tests/pytorch_tests/test_tuple_construct.py:1]:
```suggestion
# Copyright (C) 2018-2023 Intel Corporation
```

File: src/frontends/pytorch/src/transforms/prim_tuple_construct_replacer.cpp [hunk 2]
```diff
@@ -29,7 +29,19 @@ bool DecomposeTupleResults::run_on_model(const std::shared_ptr<Model>& model) {
         }
         auto inputs = input_node->inputs();
         for (auto input : inputs) {
-            model->add_results({std::make_shared<opset10::Result>(input.get_source_output())});
+            auto out = input.get_source_output();
+            auto list_construct = cast_fw_node(out.get_node_shared_ptr(), "prim::ListConstruct");
```
Comment (mvafin) [inline src/frontends/pytorch/src/transforms/prim_tuple_construct_replacer.cpp:33]: Question: if we do not do this, after we decompose `TupleConstruct` we will have `ListConstruct` on the output, can we decompose `ListConstruct` independently of `TupleConstruct`? If we had model that have `ListConstruct` instead of `TupleConstruct` on the output that will allow it to convert too

File: tests/layer_tests/pytorch_tests/test_min_max.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def forward(self, x: float, y: float):
 
         class prim_max_2_list_values(torch.nn.Module):
             def forward(self, x: float, y: float):
-                return max([x, x + y], [y, y - x])
+                return torch.tensor(max([x, x + y], [y, y - x]))
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_min_max.py:95]: Why this change become required?

File: tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py [hunk 4]
```diff
@@ -103,6 +103,9 @@ def _test(self, model, ref_net, kind, ie_device, precision, ir_version, infer_ti
             # if None is at output we skip it
             if res_item is None:
                 continue
+            if isinstance(res_item, list):
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py:106]: Need more comments here, we want the case (x, None, [y, None, z], None) unpack as [x, y, z]

**Review by mvafin — CHANGES_REQUESTED** [review]
Could we instead of adding 2 identical transformations just extend the existing ones?
* MaxPrimListConstructReplacer -> MinMaxPrimListConstructReplacer
* DecomposeTupleResults -> DecomposeTupleListResults

File: src/frontends/pytorch/src/transforms/prim_list_construct_replacer.cpp [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    ResultVector results = model->get_results();
+
+    for (size_t i = 0; i < results.size(); ++i) {
+        auto result = results[i];
+        auto input_node = result->get_input_node_shared_ptr(0);
+        auto tuple_construct = cast_fw_node(input_node, "prim::ListConstruct");
+        if (!tuple_construct) {
+            continue;
+        }
```
Comment (mvafin) [inline src/frontends/pytorch/src/transforms/prim_list_construct_replacer.cpp:28]:
```suggestion
        auto list_construct = cast_fw_node(input_node, "prim::ListConstruct");
        auto tuple_construct = cast_fw_node(input_node, "prim::TupleConstruct");
        if (!tuple_construct && !list_construct) {
            continue;
        }
```

File: src/frontends/pytorch/src/transforms/min_prim_list_construct_replacer.cpp [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+namespace pass {
+
+MinPrimListConstructReplacer::MinPrimListConstructReplacer() {
+    auto min_op = ov::pass::pattern::wrap_type<ov::op::util::FrameworkNode>();
+
+    ov::matcher_pass_callback callback = [](ov::pass::pattern::Matcher& m) {
+        auto min_op = cast_fw_node(m.get_match_root(), "prim::min");
+        if (!min_op) {
+            return false;
+        }
```
Comment (mvafin) [inline src/frontends/pytorch/src/transforms/min_prim_list_construct_replacer.cpp:31]:
```suggestion
        auto min_op = cast_fw_node(m.get_match_root(), "prim::min");
        auto max_op = cast_fw_node(m.get_match_root(), "prim::max");
        if (!min_op && !max_op) {
            return false;
        }
```

And later to figure out if you are working with min or max you can always use `if (min_op)`
